src/base/static/components/input-form/index.js

```diff
@@ -0,0 +1,551 @@
+          "Error: Unable to handle form field callback action:",
+          action
+        );
+        break;
+    }
+  }
+
+  setActiveCategoryState(categoryConfig) {
+    this.resetCategoryData();
+
+    let newFields = ImmutableMap();
+    categoryConfig.fields
+      .filter(field => {
+        if (field.type === constants.MAP_DRAWING_TOOLBAR_TYPENAME) {
+          this.customGeometryFieldname = field.name;
+        }
+
+        return (
+          this.resolveField(field).type !== constants.SUBMIT_FIELD_TYPENAME
+        );
+      })
+      .forEach(field => {
+        newFields = newFields.set(field.name, this.getInitialFieldState(field));
+        if (field.type === constants.RICH_TEXTAREA_FIELD_TYPENAME) {
+          this.richTextFields.push(field.name);
+        }
+      });
+
+    this.setState({
+      fields: newFields,
+      selectedCategory: categoryConfig.category,
+      isCategoryMenuCollapsed: true,
+      isFormSubmitting: false,
+      formValidationErrors: new Set(),
+      showValidityStatus: false,
+    });
+  }
+
+  getValidator(fieldConfig) {
+    if (fieldConfig.type === constants.SUBMIT_FIELD_TYPENAME) {
+      return {
+        validate: mayHaveAnyValue,
+        message: "",
+      };
+    } else if (fieldConfig.type === constants.MAP_DRAWING_TOOLBAR_TYPENAME) {
+      return {
+        validate: mustHaveSomeValue,
+        message: messages.missingGeometry,
+      };
+    } else {
+      return {
+        validate: fieldConfig.optional ? mayHaveAnyValue : mustHaveSomeValue,
+        message: messages.missingRequired,
+      };
+    }
+  }
+
+  getInitialFieldState(fieldConfig) {
+    // "autofill" is a better term than "autocomplete" for this feature.
+    // TODO: Update this throughout the codebase.
+    let autofillValue = fieldConfig.autocomplete
+      ? Util.getAutocompleteValue(this.resolveField(fieldConfig).name)
+      : null;
+
+    fieldConfig.hasAutofill = !!autofillValue;
+
+    let fieldValue;
+    switch (fieldConfig.type) {
+      case constants.BIG_TOGGLE_FIELD_TYPENAME:
+        fieldValue =
+          autofillValue ||
+          fieldConfig.default_value ||
+          fieldConfig.content[1].value; // "off" position of the toggle
+        break;
+      case constants.BIG_CHECKBOX_FIELD_TYPENAME:
+        fieldValue =
+          fromJS(autofillValue) ||
+          fromJS(fieldConfig.default_value) ||
+          ImmutableList();
+        break;
+      case constants.PUBLISH_CONTROL_TOOLBAR_TYPENAME:
+        fieldValue = autofillValue || fieldConfig.default_value;
+        break;
+      case constants.MAP_DRAWING_TOOLBAR_TYPENAME:
+        fieldValue = null;
+        break;
+      case constants.COMMON_FORM_ELEMENT_TYPENAME:
+        const commonFormElementConfig = Object.assign(
+          {},
+          this.props.placeConfig.common_form_elements[fieldConfig.name],
+          { name: fieldConfig.name }
+        );
+        return this.getInitialFieldState(commonFormElementConfig);
+        break;
```

Comment:
unreachable break.

Does you linter not highlight this for you?

Reply:
My linter catches this, but this seems to be a case where the break is actually necessary, I think because this case statement makes a recursive function call?

Reply:
I don't see how that's possible. After return, the function exits. How is `break` not unreachable code?

Reply:
Nevermind-- I was misremembering how this works.